File: takkernel/engine/src/main/java/com/atakmap/map/formats/c3dt/TileRenderContent.java
package com.atakmap.map.formats.c3dt;

import com.atakmap.interop.Pointer;

import gov.tak.api.annotation.DontObfuscate;

@DontObfuscate
final class TileRenderContent
{
    private final Pointer pointer;

    TileRenderContent(Pointer ptr)
    {
        this.pointer = ptr;
    }

    public Model getModel()
    {
        return getModel(pointer.raw);
    }

    private static native Model getModel(long ptr);
}

File: takkernel/engine/src/main/cpp/formats/c3dt/TileRenderContent.cpp
#include "TileRenderContent.h"

CesiumGltf::Model* TAK::Engine::Formats::Cesium3DTiles::TileRenderContent_getModel(const Cesium3DTilesSelection::TileRenderContent* tileRenderContent)
{
    return new CesiumGltf::Model(tileRenderContent->getModel());
}
